samples/bookinfo/src/details/details.rb

```diff
@@ -63,4 +75,46 @@ def get_book_details(id)
     }
 end
 
+def fetch_details_from_external_service(isbn, id)
+    uri = URI.parse('https://www.googleapis.com/books/v1/volumes?q=isbn:' + isbn)
+    http = Net::HTTP.new(uri.host, uri.port)
+    http.read_timeout = 5 # seconds
+
+    unless ENV['WITH_ISTIO'] === 'true' then
+      http.use_ssl = true
+    end
```

Comment:
What does this do? It seems that you are passing `WITH_ISTIO: true` in the deployment, so the `unless` evaluates to `false`. So it's a NO-OP, right? The env variable name, `WITH_ISTIO`, doesn't have an obvious meaning, unless I'm just not understanding.

Reply:
`WITH_ISTIO` means that the code runs with istio sidecar proxy injected in the same pod. When it is injected, it is no-op. To run the code without Istio sidecar proxy injected, the user can undefine `WITH_ISTIO`.

Possible alternative names for `WITH_ISTIO`: `WITH_ISTIO_INJECTED`, `WITH_ISTIO_SIDECAR_INJECTED`, `ISTIOFIED`. I am open to any name proposals.

Reply:
Maybe just add a comment in the code to explain why `http.use_ssl` needs to be set when running without Istio, but it isn't needed when Istio sidecars are injected.

Reply:
@frankbu I have added the comment.